src/pages/objkt-display/index.js

```diff
@@ -28,6 +28,21 @@ export const ObjktDisplay = () => {
 
   const address = context.acc?.address
 
+  var roles = []
+  function checkRoles(obj,addr) {
+    if (
+      Object.keys(obj.owners).length > 0 &&
```

Comment:
its easier to understand. the only issue is that the owners sometimes comes empty. And @crzypatchwork prefers to display it for everyone than risk not showing the swap (enabling the 2nd market) if that api fails.

Reply:
Sounds like HIC should calculate the owners itself from chain, will lessen reliance on timely data from upstream provider.

This is not just a refactor, it does correct the bug that swap tab was only showing to creator, when API was showing correct data.

I'll close for now, but I recommend it would be needed in future for usability, as it would help remove confusion from non-owners - the appearance to them of swap and burn buttons has been reported as a bug in Discord. Happy to resubmit when the API reliability is increased.